OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -178,21 +178,52 @@ static class DreamProcNativeRoot {
         }
 
         [DreamProc("block")]
-        [DreamProcParameter("Start", Type = DreamValueType.DreamObject)]
-        [DreamProcParameter("End", Type = DreamValueType.DreamObject)]
+        [DreamProcParameter("Start", Type = DreamValueType.DreamObject | DreamValueType.Float)]
+        [DreamProcParameter("End", Type = DreamValueType.DreamObject | DreamValueType.Float)]
+        [DreamProcParameter("StartZ", Type = DreamValueType.Float)]
+        [DreamProcParameter("EndX", Type = DreamValueType.Float)]
+        [DreamProcParameter("EndY", Type = DreamValueType.Float)]
+        [DreamProcParameter("EndZ", Type = DreamValueType.Float)]
         public static DreamValue NativeProc_block(NativeProc.State state) {
-            if (!state.GetArgument(0, "Start").TryGetValueAsDreamObjectOfType(state.ObjectTree.Turf, out var start))
-                return new DreamValue(state.ObjectTree.CreateList());
-            if (!state.GetArgument(1, "End").TryGetValueAsDreamObjectOfType(state.ObjectTree.Turf, out var end))
-                return new DreamValue(state.ObjectTree.CreateList());
+            int x1;
+            int y1;
+            int z1;
+            int x2;
+            int y2;
+            int z2;
+            if (state.Arguments.Count <= 2) { // Is this the turf-style call?
```

Comment:
Using named arguments (`block(Start=..., End=...)`) will lead there to be 6 arguments because it'll fill the other four with null. It may be better to instead check if the first arguments is a turf.